### PR TITLE
Propose an issue triaging process for maplibre projects

### PR DESCRIPTION
**Pull request by hy9be**
Moving the proposal from my local repo to here get collect more feedback: https://github.com/hy9be/maplibre-gl-native/pull/1

Created this document under a new folder named ``process``. This folder can be the home for future docs for roadmap process, release process, and etc.

### CODE REVIEW

**Comment by hy9be** [comment]
FYI @HarelM @ntadej @roblabs @lseelenbinder @birkskyum @Keith @Joxit @klokan @mojodna @Pessimistress @prozessor13

**Comment by wipfli** [comment]
Thanks for taking the time to prepare this document. Having the labels and associated reasons to close an issue is a great idea I find.

I think it would be better to not distinguish between the Technical Steering Committee and the Community. I would rather say that people with write access can handle adding labels, closing issues, etc.

So far feature requests by people had to be driven by them. We never promised to implement something if someone asked for it I think. Maybe we can change this now, but his is just how it was...

**Comment by wipfli** [comment]
What I really like about MapLibre is that it is not we and them. The community is MapLibre... Sorry if I don't find the right words here...

File: process/Issue_Triaging.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+|First Step     |Next Step            |Result     |
+|---            |---                  |---        |
+|TSC review ->  |Accepted, or         |
+|               |Community review ->  |Closed, or |
+|               |                     |Accepted   |
+
+Please note:
+1. TSC reviewers can accept an issue but cannot close an issue before it is being reviewed by the community (the "Community Review" step).
+2. We are much less likely to accept PRs that have no associated accepted issues
```
Comment (ntadej) [inline process/Issue_Triaging.md:23]: I see no reason why PRs would not be accepted if they are reasonable. Maybe we should reword this to say that it's good to open issues if you plan to work on something to make sure you get feedback in advance and to avoid duplicating work.

File: process/Issue_Triaging.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+Issues can be at four major states during this process. They are easily identifiable:
+
+|State|What your GitHub issue looks like|
+|---|---|
+|TSC Review|matches the query `label:"tsc-review"`|
```
Comment (ntadej) [inline process/Issue_Triaging.md:29]: `tsc-review` may be confusing to new users. Could it be just `In Review`?

File: process/Issue_Triaging.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+|Type|Description|
+|---|---|
+|[`needs-more-info`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | not possible to assign a type label due to missing information|
+|[`bug`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | the implementation of a feature is not correct|
+|[`feature-request`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | request for a new feature|
+|[`tech-debt`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | improve the implementation/architecture|
+|[`engineering`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | issues related to our engineering system or our processes|
+|[`polish`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | a feature could be improved, but not necessarily a bug|
+|[`*question`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | we should direct questions to StackOverflow|
```
Comment (ntadej) [inline process/Issue_Triaging.md:53]: We also have discussions available.

Do we need to triage also discussions? 😊

File: process/Issue_Triaging.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+### Feature Areas Labels
+
+Each issue must have a **feature area** label.
```
Comment (ntadej) [inline process/Issue_Triaging.md:57]: This can get tricky. In native we also have SDK features so it may be hard to cover all areas with labels.

**Review by ntadej — CHANGES_REQUESTED** [review]

**Comment by ntadej** [comment]
Thanks also from my side.

As you hint it in the text, it may be good to clarify if we should also use milestones/boards.

File: process/Issue_Triaging.md (hunk 1 above)
Comment (HarelM) [reply process/Issue_Triaging.md:23]: I would expect to see an issue in order to define a problem statement and a PR to implement the solution. Having said that, if this is small a PR in a lot of cases is enough.

File: process/Issue_Triaging.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+Each issue must have a **type** label. Most type labels are grey, some are yellow. Bugs are grey with a touch of red.
+
+|Type|Description|
+|---|---|
+|[`needs-more-info`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | not possible to assign a type label due to missing information|
+|[`bug`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | the implementation of a feature is not correct|
+|[`feature-request`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | request for a new feature|
+|[`tech-debt`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | improve the implementation/architecture|
+|[`engineering`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | issues related to our engineering system or our processes|
```
Comment (HarelM) [inline process/Issue_Triaging.md:51]: Seems redundant to me, tech-debt seems enough...

File: process/Issue_Triaging.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+|[`polish`](https://github.com/maplibre/maplibre-gl-native/labels/xx) | a feature could be improved, but not necessarily a bug|
```
Comment (HarelM) [inline process/Issue_Triaging.md:52]: Either feature request or bug, I don't think this is needed.

File: process/Issue_Triaging.md (hunk 3 above)
Comment (HarelM) [reply process/Issue_Triaging.md:53]: No. An issue that is a question should be converted to discussion, IMHO.

File: process/Issue_Triaging.md (hunk 4 above)
Comment (HarelM) [reply process/Issue_Triaging.md:57]: Yea, I don't find this appealing as well...

File: process/Issue_Triaging.md [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+If issues are suitable for beginners we may add the `good-first-issue` label and we add code pointers that help beginners to get started with a PR.
+
+Sometime, we get issues that we can't or don't have the time to reproduce due to the complexity or time requirements of the setup but that we indeed suspect to be issues. We label those issues with `investigation-wanted`. What we are looking for is help in reproducing and analyzing the issue.
+
+
+## Community Review
+
+### Accepting issue
+Community members can "up-vote" an issue (adding a GitHub +1/"👍" reaction to the issue description). If an issue surpasses the 20 up-votes in 60 days, the issue will be accepted.
```
Comment (HarelM) [inline process/Issue_Triaging.md:98]: I'm not sure what accepted means...

File: process/Issue_Triaging.md [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+The basic process of triaging issues is
+
+|First Step     |Next Step            |Result     |
+|---            |---                  |---        |
+|TSC review ->  |Accepted, or         |
+|               |Community review ->  |Closed, or |
+|               |                     |Accepted   |
+
+Please note:
+1. TSC reviewers can accept an issue but cannot close an issue before it is being reviewed by the community (the "Community Review" step).
```
Comment (HarelM) [inline process/Issue_Triaging.md:22]: This seems too strict...

**Comment by HarelM** [comment]
I think this is a great start! Thanks for sharing! Having said that, I find very low possibility fir correlation to reality in the current state of the project and issues maintenance... :-(

**Comment by HarelM** [comment]
Just an FYI, I went over all the issues in maplibre-gl and categorized them - bugs, enhancements, discussions, things that are not clear or need more info etc.
There's still a good question of what's next.
Currently the priority is decided by people who would like to contribute and create a PR for an existing issue as I feel there's not enough maintainers to actually decide what's more important and focus on that.
I don't know if this will change in the future or how...

**Comment by prozessor13** [comment]
thx @HarelM, for me this will help a lot!

**Comment by nyurik** [comment]
any updates on this?
